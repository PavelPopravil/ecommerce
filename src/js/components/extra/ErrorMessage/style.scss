@import "../../../../scss/main/variables";

.error {
  box-shadow: $b-shadow;
  background-color: $body-bg-color;
  padding: 30px 25px 25px;
  font-size: $font-size-h4;
}