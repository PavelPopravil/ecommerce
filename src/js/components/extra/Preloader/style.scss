@import "../../../../scss/main/variables";

.preloader {
  width: 100%;

  &__spinner {
    border-radius: 50%;
    width: 60px;
    height: 60px;
  }

  &__spinner {
    margin: 20px auto;
    font-size: 10px;
    position: relative;
    text-indent: -9999em;
    border-top: 10px solid rgba($brand-1, 0.2);
    border-right: 10px solid rgba($brand-1, 0.2);
    border-bottom: 10px solid rgba($brand-1, 0.2);
    border-left: 10px solid $brand-1;
    transform: translateZ(0);
    animation: load8 1.1s infinite linear;

    @media (max-width: $screen-xs-max) {
      margin-top: 10px;
      margin-bottom: 10px;
    }
  }

  @-webkit-keyframes load8 {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
  @keyframes load8 {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
}

//.preloader {
//  //display: none;
//  position: absolute;
//  top: 0;
//  left: 0;
//  width: 100%;
//  height: 100%;
//  z-index: 99;
//
//  &.active {
//    //display: block;
//    background-color: rgba(255,255,255, .7);
//  }
//
//  &__spinner {
//    display: block;
//    position: absolute;
//    left: 50%;
//    top: 50%;
//    width: 150px;
//    height: 150px;
//    margin: -75px 0 0 -75px;
//    border-radius: 50%;
//    border: 3px solid transparent;
//    border-top-color: $link-color;
//    animation: spin 2s linear infinite;
//    z-index: 1001;
//
//    &:after,
//    &:before {
//      content: '';
//      position: absolute;
//      top: 5px;
//      left: 5px;
//      right: 5px;
//      bottom: 5px;
//      border-radius: 50%;
//      border: 3px solid transparent;
//      border-top-color: $link-color-hover;
//      animation: spin 3s linear infinite;
//    }
//
//    &:after {
//      top: 15px;
//      left: 15px;
//      right: 15px;
//      bottom: 15px;
//      border-top-color: $link-color;
//      animation: spin 1.5s linear infinite;
//    }
//
//    &.xs {
//      $__preloader-xs-size: 40px;
//      width: $__preloader-xs-size * 2;
//      height: $__preloader-xs-size * 2;
//      margin: (-$__preloader-xs-size) 0 0 (-$__preloader-xs-size);
//    }
//
//    &.sm {
//      $__preloader-sm-size: 50px;
//      width: $__preloader-sm-size * 2;
//      height: $__preloader-sm-size * 2;
//      margin: (-$__preloader-sm-size) 0 0 (-$__preloader-sm-size);
//    }
//
//    &.md {
//      $__preloader-md-size: 65px;
//      width: $__preloader-md-size * 2;
//      height: $__preloader-md-size * 2;
//      margin: (-$__preloader-md-size) 0 0 (-$__preloader-md-size);
//    }
//  }
//}
//
//@keyframes spin {
//  0% {
//    transform: rotate(0deg)
//  }
//  to {
//    transform: rotate(360deg)
//  }
//}