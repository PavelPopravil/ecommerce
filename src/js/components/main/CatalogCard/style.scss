@import "../../../../scss/main/variables";

.catalog-card {
  flex: 1;
  max-width: 100%;
  padding: 25px 20px;
  box-shadow: $b-shadow;
  text-decoration: none;

  &:hover,
  &:focus {
    box-shadow: $b-shadow-hover;
  }

  &__pic-wrap {
    height: 180px;
    margin-bottom: 15px;
  }

  &__pic {
    margin: 0 auto;
    max-height: 100%;
  }

  &__text-wrap {}

  &__title {}

  &__price {
    font-size: 16px;
    color: $link-color;
    font-family: $font-m;
  }
}