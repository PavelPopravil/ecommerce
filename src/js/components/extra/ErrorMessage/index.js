import React from 'react';
import './style.scss';

export default (props) => {
  return (
    <div className='error'>{props.children}</div>
  )
}