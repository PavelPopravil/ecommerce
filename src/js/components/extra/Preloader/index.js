import React from 'react';
import './style.scss';

export default () => {
  console.log('render Preloader');

  return (
    <div className='preloader'><div className='preloader__spinner'> </div></div>
  );
};