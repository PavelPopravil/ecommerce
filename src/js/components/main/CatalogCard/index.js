import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

import './style.scss';

class CatalogCard extends React.PureComponent {

  static propTypes = {
    'pic': PropTypes.string.isRequired,
    'title': PropTypes.string.isRequired,
    'price': PropTypes.string.isRequired
  };

  render() {
    console.log('render CatalogCard');
    const {pic, title, price} = this.props;

    return (
      <Link className='catalog-card' to={'/about'}>
        <div className='catalog-card__pic-wrap'>
          <img className='catalog-card__pic' src={`${process.env.PUBLIC_URL}/img/${pic}`} alt={title} />
        </div>
        <div className='catalog-card__text-wrap'>
          <div className="h3 catalog-card__title">{title}</div>
          <div className="catalog-card__price">{price}</div>
        </div>
      </Link>
    )
  }
}

export default CatalogCard;