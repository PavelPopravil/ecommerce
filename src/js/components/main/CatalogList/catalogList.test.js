import React from 'react';
import {shallow} from 'enzyme';

import {CatalogList} from './index';
import CatalogCard from '../CatalogCard/index';
import Preloader from '../../extra/Preloader/index';
import ErrorMessage from '../../extra/ErrorMessage/index';

describe('CatalogList render', () => {

  const props = {
    isLoading: false,
    data: [],
    errorMsg: null,
    onFetchCatalog: () => {}
  };

  describe('CatalogList initial state', () => {

    const mockFn = jest.fn();

    const newProps = {
      ...props,
      onFetchCatalog: mockFn
    };

    const catalogContainer = shallow(<CatalogList {...newProps} />);

    it('renders properly', () => {
      expect(catalogContainer).toMatchSnapshot();
    });

    it('should recive function from redux and call it once', () => {
      expect(mockFn).toHaveBeenCalledTimes(1);
    });

  });

  describe('CatalogList fetch data start', () => {

    const newProps = {
      ...props,
      isLoading: true
    };

    const catalogContainer = shallow(<CatalogList {...newProps} />);

    it('Should render Preloader component', () => {
      expect(catalogContainer.find(Preloader)).toHaveLength(1);
    });

    it('Shouldn\'t render CatalogCard component', () => {
      expect(catalogContainer.find(CatalogCard)).toHaveLength(0);
    });

    it('Shouldn\'t render div with error message', () => {
      expect(catalogContainer.find('.failure-block')).toHaveLength(0);
    });

    it('renders properly', () => {
      expect(catalogContainer).toMatchSnapshot()
    })
  });

  describe('CatalogList fetch data success', () => {

    const newProps = {
      ...props,
      isLoading: false,
      data: [
        {
          mod: "big",
          id: 1,
          pic: "img/catalog_imac.png",
          path: 'mac',
          title: "Mac",
          price: 1650
        }
      ]
    };

    const catalogContainer = shallow(<CatalogList {...newProps} />);

    it('Shouldn\'t render Preloader component', () => {
      expect(catalogContainer.find(Preloader)).toHaveLength(0);
    });

    it('Shouldn\'t render div with error message', () => {
      expect(catalogContainer.find('.failure-block')).toHaveLength(0);
    });

    it('Should render CatalogCard component', () => {
      expect(catalogContainer.find(CatalogCard)).toHaveLength(newProps.data.length);
    });

    it('renders properly', () => {
      expect(catalogContainer).toMatchSnapshot()
    })

  });

  describe('CatalogList fetch data failure', () => {

    const newProps = {
      ...props,
      isLoading: false,
      errorMsg: 'fetch error'
    };

    const catalogContainer = shallow(<CatalogList {...newProps} />);

    it('Shouldn\'t render Preloader component', () => {
      expect(catalogContainer.find(Preloader)).toHaveLength(0);
    });

    it('Shouldn\'t render CatalogCard component', () => {
      expect(catalogContainer.find(CatalogCard)).toHaveLength(0);
    });

    it('Should render ErrorMessage component', () => {
      expect(catalogContainer.find(ErrorMessage)).toHaveLength(1);
    });

    it('renders properly', () => {
      expect(catalogContainer).toMatchSnapshot()
    })

  });
});