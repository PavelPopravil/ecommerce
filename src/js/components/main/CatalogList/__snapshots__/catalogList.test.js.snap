// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogList render CatalogList fetch data failure renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CatalogList
    data={Array []}
    errorMsg="fetch error"
    isLoading={false}
    onFetchCatalog={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="catalog-list__title"
        >
          Каталог
        </h2>,
        <div
          className="catalog-list__list"
        >
          <div
            className="row"
          >
            <div
              className="col-12 failure-block"
            >
              Произошла ошибка fetch error
            </div>
          </div>
        </div>,
      ],
      "className": "catalog-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Каталог",
          "className": "catalog-list__title",
        },
        "ref": null,
        "rendered": "Каталог",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-12 failure-block"
            >
              Произошла ошибка fetch error
            </div>
          </div>,
          "className": "catalog-list__list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-12 failure-block"
            >
              Произошла ошибка fetch error
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Произошла ошибка fetch error",
              "className": "col-12 failure-block",
            },
            "ref": null,
            "rendered": "Произошла ошибка fetch error",
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="catalog-list__title"
          >
            Каталог
          </h2>,
          <div
            className="catalog-list__list"
          >
            <div
              className="row"
            >
              <div
                className="col-12 failure-block"
              >
                Произошла ошибка fetch error
              </div>
            </div>
          </div>,
        ],
        "className": "catalog-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Каталог",
            "className": "catalog-list__title",
          },
          "ref": null,
          "rendered": "Каталог",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="col-12 failure-block"
              >
                Произошла ошибка fetch error
              </div>
            </div>,
            "className": "catalog-list__list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-12 failure-block"
              >
                Произошла ошибка fetch error
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Произошла ошибка fetch error",
                "className": "col-12 failure-block",
              },
              "ref": null,
              "rendered": "Произошла ошибка fetch error",
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CatalogList render CatalogList fetch data start renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CatalogList
    data={Array []}
    errorMsg={null}
    isLoading={true}
    onFetchCatalog={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="catalog-list__title"
        >
          Каталог
        </h2>,
        <div
          className="catalog-list__list"
        >
          <div
            className="row"
          >
            <_default />
          </div>
        </div>,
      ],
      "className": "catalog-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Каталог",
          "className": "catalog-list__title",
        },
        "ref": null,
        "rendered": "Каталог",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <_default />
          </div>,
          "className": "catalog-list__list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <_default />,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="catalog-list__title"
          >
            Каталог
          </h2>,
          <div
            className="catalog-list__list"
          >
            <div
              className="row"
            >
              <_default />
            </div>
          </div>,
        ],
        "className": "catalog-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Каталог",
            "className": "catalog-list__title",
          },
          "ref": null,
          "rendered": "Каталог",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <_default />
            </div>,
            "className": "catalog-list__list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <_default />,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CatalogList render CatalogList fetch data success renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CatalogList
    data={
      Array [
        Object {
          "id": 1,
          "mod": "big",
          "path": "mac",
          "pic": "img/catalog_imac.png",
          "price": "от 1 650 руб",
          "title": "Mac",
        },
      ]
    }
    errorMsg={null}
    isLoading={false}
    onFetchCatalog={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="catalog-list__title"
        >
          Каталог
        </h2>,
        <div
          className="catalog-list__list"
        >
          <div
            className="row"
          >
            <div
              className="catalog-list__item col-lg-6 col-md-6 col-12"
            >
              <CatalogCard
                path="mac"
                pic="img/catalog_imac.png"
                price="от 1 650 руб"
                title="Mac"
              />
            </div>
          </div>
        </div>,
      ],
      "className": "catalog-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Каталог",
          "className": "catalog-list__title",
        },
        "ref": null,
        "rendered": "Каталог",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="catalog-list__item col-lg-6 col-md-6 col-12"
            >
              <CatalogCard
                path="mac"
                pic="img/catalog_imac.png"
                price="от 1 650 руб"
                title="Mac"
              />
            </div>
          </div>,
          "className": "catalog-list__list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="catalog-list__item col-lg-6 col-md-6 col-12"
              >
                <CatalogCard
                  path="mac"
                  pic="img/catalog_imac.png"
                  price="от 1 650 руб"
                  title="Mac"
                />
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <CatalogCard
                  path="mac"
                  pic="img/catalog_imac.png"
                  price="от 1 650 руб"
                  title="Mac"
                />,
                "className": "catalog-list__item col-lg-6 col-md-6 col-12",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "path": "mac",
                  "pic": "img/catalog_imac.png",
                  "price": "от 1 650 руб",
                  "title": "Mac",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="catalog-list__title"
          >
            Каталог
          </h2>,
          <div
            className="catalog-list__list"
          >
            <div
              className="row"
            >
              <div
                className="catalog-list__item col-lg-6 col-md-6 col-12"
              >
                <CatalogCard
                  path="mac"
                  pic="img/catalog_imac.png"
                  price="от 1 650 руб"
                  title="Mac"
                />
              </div>
            </div>
          </div>,
        ],
        "className": "catalog-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Каталог",
            "className": "catalog-list__title",
          },
          "ref": null,
          "rendered": "Каталог",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="catalog-list__item col-lg-6 col-md-6 col-12"
              >
                <CatalogCard
                  path="mac"
                  pic="img/catalog_imac.png"
                  price="от 1 650 руб"
                  title="Mac"
                />
              </div>
            </div>,
            "className": "catalog-list__list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="catalog-list__item col-lg-6 col-md-6 col-12"
                >
                  <CatalogCard
                    path="mac"
                    pic="img/catalog_imac.png"
                    price="от 1 650 руб"
                    title="Mac"
                  />
                </div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": <CatalogCard
                    path="mac"
                    pic="img/catalog_imac.png"
                    price="от 1 650 руб"
                    title="Mac"
                  />,
                  "className": "catalog-list__item col-lg-6 col-md-6 col-12",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "path": "mac",
                    "pic": "img/catalog_imac.png",
                    "price": "от 1 650 руб",
                    "title": "Mac",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CatalogList render CatalogList initial state renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CatalogList
    data={Array []}
    errorMsg={null}
    isLoading={false}
    onFetchCatalog={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="catalog-list__title"
        >
          Каталог
        </h2>,
        <div
          className="catalog-list__list"
        >
          <div
            className="row"
          />
        </div>,
      ],
      "className": "catalog-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Каталог",
          "className": "catalog-list__title",
        },
        "ref": null,
        "rendered": "Каталог",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          />,
          "className": "catalog-list__list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="catalog-list__title"
          >
            Каталог
          </h2>,
          <div
            className="catalog-list__list"
          >
            <div
              className="row"
            />
          </div>,
        ],
        "className": "catalog-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Каталог",
            "className": "catalog-list__title",
          },
          "ref": null,
          "rendered": "Каталог",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            />,
            "className": "catalog-list__list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
